snow/validators/overridden_manager.go

```diff
@@ -0,0 +1,111 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package validators
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/crypto/bls"
+	"github.com/ava-labs/avalanchego/utils/formatting"
+	"github.com/ava-labs/avalanchego/utils/set"
+)
+
+var _ Manager = (*manager)(nil)
+
+// NewOverriddenManager returns a Manager that overrides of all calls to the underlying Manager
+// to only operate on the validators in [subnetID].
+func NewOverriddenManager(subnetID ids.ID, manager Manager) Manager {
+	return &overriddenManager{
+		subnetID: subnetID,
+		manager:  manager,
+	}
+}
+
+// overriddenManager is a wrapper around a Manager that overrides of all calls to the underlying Manager
+// to only operate on the validators in [subnetID].
+// subnetID here is typically the primary network ID, as it has the superset of all subnet validators.
+type overriddenManager struct {
+	manager  Manager
+	subnetID ids.ID
+}
+
+func (o *overriddenManager) AddStaker(_ ids.ID, nodeID ids.NodeID, pk *bls.PublicKey, txID ids.ID, weight uint64) error {
+	return o.manager.AddStaker(o.subnetID, nodeID, pk, txID, weight)
+}
+
+func (o *overriddenManager) AddWeight(_ ids.ID, nodeID ids.NodeID, weight uint64) error {
+	return o.manager.AddWeight(o.subnetID, nodeID, weight)
+}
+
+func (o *overriddenManager) GetWeight(_ ids.ID, validatorID ids.NodeID) uint64 {
```

Comment:
Can we be consistent with the naming? (I think I prefer `nodeID` over `validatorID`)

Reply:
done